hack/verify-e2e-test-ownership.sh

```diff
@@ -0,0 +1,159 @@
+      line: \$code[\$i] | "\(.FileName):\(.LineNumber)",
+    } + (\$code[\$i] | .FullStackTrace | match("^(.*)\\\.(.+)\\\(.*\\n") | .captures | {
+        package: .[0].string,
+        func: .[1].string
+    })],
+  } | {
+    owner: .calls[0].sig,
+    full_name: .calls | map(.text) | join(" "),
+    calls,
+    policies: [(
+      .calls[0] |
+        {
+          fail: (.sig == "unknown"),
+          level: "FAIL",
+          category: "unowned_test",
+          reason: "should start with [sig-foo]",
+          found: .,
+        }, {
+          fail: (.func != "SIGDescribe"),
+          level: "WARN",
+          category: "no_sig_describe",
+          reason: "should use top-level SIGDescribe",
+          found: .,
+        }
+      ), (
+      .calls[1:] |
+        (map(select(.sig != "unknown")) // [] | {
+          fail: . | any,
+          level: "WARN",
+          category: "too_many_sigs",
+          reason: "should not have multiple [sig-foo] tags",
+          found: .,
+        }),
+        (map(select(.func == "SIGDescribe")) // [] | {
+          fail: . | any,
+          level: "WARN",
+          category: "nested_sig_describe",
+          reason: "should not use nested SIGDescribe",
+          found: .,
+        })
+      ), (
+      .calls |
+        (map(select(.text | match("\\\[k8s\\\.io\\\]"))) // [] | {
+          fail: . | any,
+          level: "WARN",
+          category: "has_k8s_io",
+          reason: "should not have [k8s.io] in test names",
+          found: .,
+        }),
+        (map(select(.func == "KubeDescribe")) // [] | {
+          fail: . | any,
+          level: "WARN",
+          category: "calls_kube_describe",
+          reason: "should not have any KubeDescribe calls",
+          found: .,
+        })
+      )
+    ]
+}]
+EOS
+
+summary_jq=${tmpdir}/summary.jq
+summary_json=${tmpdir}/summary.json
+cat >${summary_jq} <<EOS
+. as \$results |
+reduce \$results[0].policies[] as \$p ({}; . + {
+  (\$p.category): \$results | map(. + {policy: .policies[] | select(.category == \$p.category)}) | {
+    level: \$p.level,
+    reason: \$p.reason,
+    passing: map(select(.policy.fail | not)) | length,
+    failing: map(select(.policy.fail)) | length,
+    failures: map(select(.policy.fail) | .full_name),
+  }
+}) + {
+  all_policies: \$results | {
+    level: "WARN",
+    reason: "passing all policies",
+    passing: map(select(.policies | map(.fail) | any | not)) | length,
+    failing: map(select(.policies | map(.fail) | any)) | length,
+    failures: map(select(.policies | map(.fail) | any) | .full_name),
+  }
+} | with_entries(.value += { log: (if (.value.failing == 0) then "PASS" else .value.level end) })
+
+EOS
+
+# cat -n "${results_jq}"
+<"${spec_summaries}" jq --slurp --from-file "${results_jq}" > "${results_json}"
+# cat -n "${summary_jq}"
+<"${results_json}" jq --from-file "${summary_jq}" > "${summary_json}"
+
+summary_sh="${tmpdir}/summary.sh"
```

Comment:
What would be useful is if requested a DEBUG mode, to also print the tests that are actually failing particular checks.

Reply:
Agree. I'm planning to add an alternate mode that dumps all failures but on a per-test basis

The output is really verbose right now, but it could be we just decide that's an accurate representation of the current state